### ISSUE TITLE
preresolve types resolves string array incorrectly

### ISSUE THREAD

**Issue by mlabrum** [dotansimha/graphql-code-generator#2211]
**Describe the bug**
preresolve types resolves string array incorrectly as `string` rather than `string[]`

**To Reproduce**

https://codesandbox.io/s/graphql-codegen-issue-template-wbzsk?fontsize=14

1. My GraphQL schema:

```graphql
schema {
  query: Query
}

type Query {
  test: [String!]!
}

```

2. My GraphQL operations:

```graphql
query test{
 	test 
}
```

3. My `codegen.yml` config file:

```yml
schema: myschema.graphql
documents: mydocument.graphql
generates:
  types.ts:
    plugins:
      - typescript
      - typescript-operations
    config:
      preResolveTypes: true

```

**Current behavior**
```typescript
export type Maybe<T> = T | null;
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: string;
  String: string;
  Boolean: boolean;
  Int: number;
  Float: number;
};

export type Query = {
  __typename?: "Query";
  test: Array<Scalars["String"]>;
};
export type TestQueryVariables = {};

export type TestQuery = { __typename?: "Query"; test: string };
```

**Expected behavior**
```typescript
export type Maybe<T> = T | null;
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: string;
  String: string;
  Boolean: boolean;
  Int: number;
  Float: number;
};

export type Query = {
  __typename?: "Query";
  test: Array<Scalars["String"]>;
};
export type TestQueryVariables = {};

export type TestQuery = { __typename?: "Query"; test: Array<string> };
```

**Environment:**
( See the code sandbox )

**Comment by dotansimha** [comment]
Fixed in https://github.com/dotansimha/graphql-code-generator/pull/2270

**Comment by dotansimha** [comment]
Fixed in 1.5.0 🎉
